### PR TITLE
Use native azdo command for checkout

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

File: eng/pipelines/checkout-unix-task.yml [hunk 1]
```diff
@@ -1,11 +1,5 @@
...
-  - script: |
-      set -x
-      git init
-      git remote add origin "$(Build.Repository.Uri)"
-      git fetch --progress --no-tags --depth=1 origin "$(Build.SourceVersion)"
-      git checkout "$(Build.SourceVersion)"
-    displayName: Shallow Checkout
+  - checkout: self
+    clean: true
+    fetchDepth: 1 # Fetch only one commit
```
Comment (JoeRobich) [inline eng/pipelines/checkout-unix-task.yml:5]: Glad they added this.

File: eng/pipelines/checkout-unix-task.yml [hunk 2]
```diff
@@ -1,11 +1,5 @@
...
-  - checkout: none
-
-  - script: |
-      set -x
-      git init
-      git remote add origin "$(Build.Repository.Uri)"
-      git fetch --progress --no-tags --depth=1 origin "$(Build.SourceVersion)"
-      git checkout "$(Build.SourceVersion)"
-    displayName: Shallow Checkout
+  - checkout: self
```
Comment (adiaaida) [inline eng/pipelines/checkout-unix-task.yml:3]: I don't think 2 files are needed for this anymore given that both do the same thing now. Not sure if you want to make that many changes though

**Review by adiaaida — APPROVED** [review]

**Review by JoeRobich — APPROVED** [review]
Thanks Matt!

File: eng/pipelines/checkout-unix-task.yml (hunk 2 above)
Comment (JoeRobich) [reply eng/pipelines/checkout-unix-task.yml:3]: I am happy to clean this up and backport it.

**Review by jasonmalinowski — APPROVED** [review]

**Review by adiaaida — APPROVED** [review]

**Comment by 333fred** [comment]
Does this avoid tags as well? We have a decent number of them.

**Comment by JoeRobich** [comment]
> Does this avoid tags as well? We have a decent number of them.

Doesn't appear to be an option either - https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines

**Comment by mmitche** [comment]
> > Does this avoid tags as well? We have a decent number of them.
> 
> Doesn't appear to be an option either - https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/steps-checkout?view=azure-pipelines

I'm trying to figure out where to submit a PR or add a request. I wouldn't block on removing the manual checkout for this though.

**Comment by adiaaida** [comment]
up dooted

**Comment by RikkiGibson** [comment]
Is this solving a correctness issue? At a glance this still appears to be 3-4x slower than the shallow checkout that we do currently. That difference is less than 2 minutes, granted.

**Comment by mmitche** [comment]
> Is this solving a correctness issue? At a glance this still appears to be 3-4x slower than the shallow checkout that we do currently. That difference is less than 2 minutes, granted.

Up to you guys whether you take it. Git was removed on purpose from the images with VS on them:  https://github.com/dotnet/arcade/issues/9166

**Review by RikkiGibson — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Let's merge this to resolve the current CI failures in main.

**Pull request merged**